EM_X86_64 with R_386_GOTOFF is not supported in text relocation

I get an assertion error when trying to build an mpy module from c code. I am using the code provided in the docs (https://docs.micropython.org/en/latest/develop/natmod.html). The error is as follow:
```
Traceback (most recent call last):
  File "../../build/micropython//tools/mpy_ld.py", line 1082, in <module>
    main()
  File "../../build/micropython//tools/mpy_ld.py", line 1078, in main
    do_link(args)
  File "../../build/micropython//tools/mpy_ld.py", line 1049, in do_link
    link_objects(env, len(native_qstr_vals), len(native_qstr_objs))
  File "../../build/micropython//tools/mpy_ld.py", line 831, in link_objects
    do_relocation_text(env, sec.addr, r)
  File "../../build/micropython//tools/mpy_ld.py", line 577, in do_relocation_text
    assert 0, r_info_type
AssertionError: 9
```
And those are my settings:
```
env.arch.name is EM_X86_64
r_info_type is 9 (R_386_GOTOFF)
```

I am not sure how these GOT calculations are made. I might be able to work on it if any instructions are provided. Thanks a lot!

--- a/tools/mpy_ld.py
+++ b/tools/mpy_ld.py
@@ -73,6 +73,7 @@ R_XTENSA_SLOT0_OP = 20
 R_ARM_BASE_PREL = 25  # aka R_ARM_GOTPC
 R_ARM_GOT_BREL = 26  # aka R_ARM_GOT32
 R_ARM_THM_JUMP24 = 30
+R_X86_64_GOTPCREL = 9
 R_X86_64_REX_GOTPCRELX = 42
 R_386_GOT32X = 43
 
@@ -132,7 +133,7 @@ ARCH_DATA = {
         | MICROPY_OPT_CACHE_MAP_LOOKUP_IN_BYTECODE,
         2,
         8,
-        (R_X86_64_REX_GOTPCRELX,),
+        (R_X86_64_GOTPCREL, R_X86_64_REX_GOTPCRELX),
         asm_jump_x86,
     ),
     "armv7m": ArchData(
@@ -536,7 +537,10 @@ def do_relocation_text(env, text_addr, r):
         # Relcation pointing to GOT
         reloc = addr = env.got_entries[s.name].offset
 
-    elif env.arch.name == "EM_X86_64" and r_info_type == R_X86_64_REX_GOTPCRELX:
+    elif env.arch.name == "EM_X86_64" and r_info_type in (
+        R_X86_64_GOTPCREL,
+        R_X86_64_REX_GOTPCRELX,
+    ):
         # Relcation pointing to GOT
         got_entry = env.got_entries[s.name]
         addr = env.got_section.addr + got_entry.offset

